### DISCUSSION TITLE
opensearch.keystore not created

### DISCUSSION THREAD

**Discussion by jvlavl** [cisagov/Malcolm#515] [Troubleshooting] [answered]
**Describe the bug**
the opensearch.keystore isn't created

Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /home/analist/Malcolm/opensearch/opensearch.keystore

If applicable, attach screenshots or container logs (e.g., the relevant bits of `./scripts/logs`) to help explain your problem.

**Malcolm Version:**
 - Version [ v24.01.0]

**How are you running Malcolm?**
- [ ] [ISO installed](https://cisagov.github.io/Malcolm/docs/malcolm-hedgehog-e2e-iso-install.html#InstallationExample) (on VM or dedicated hardware)

**Additional context**

**Comment by mmguero** [comment]
You're using the startup script (`./scripts/start`) to run Malcolm, or the corresponding icons on the toolbar on the Malcolm install desktop, right? You're not trying to do `docker compose up` manually?

This should be created automatically on startup, so we need to determine why it's not happening. First, double-check and make sure the file doesn't exist incorrectly somehow (like with the wrong permissions, or as a directory, or something). Run `ls -l ./opensearch/opensearch.keystore` in your Malcolm installation directory. I assume this returns nothing? If it returns something, either a file with the wrong permissions, or an empty directory, or something else, use `sudo rm ./opensearch/opensearch.keystore` or `sudo rmdir ./opensearch/opensearch.keystore` to remove it before trying to start up.

If that doesn't fix the problem, we'll start with extra debugging turned on to figure out where the problem is. Run `./scripts/start -v`.

Pretty early on in the startup, you'll see something like this:

```
...
['docker', 'compose', '--profile', 'malcolm', '-f', '/home/analyst/Malcolm/docker-compose.yml', 'version']() returned 0: ['Docker Compose version v2.29.7', '']
Run profile (process.env): malcolm
['docker', 'compose', '--profile', 'malcolm', '-f', '/home/analyst/Malcolm/docker-compose.yml', 'ps', '-q', 'opensearch']() returned 0: []
['docker', 'run', '--rm', '--entrypoint', '/usr/local/bin/docker-uid-gid-setup.sh', '--env', 'PUID=1000', '--env', 'DEFAULT_UID=1000', '--env', 'PGID=1000', '--env', 'DEFAULT_GID=1000', '--env', 'PUSER_CHOWN=/usr/share/opensearch/config', '-v', '/home/analyst/Malcolm/opensearch:/usr/share/opensearch/config:rw', '-w', '/usr/share/opensearch/config', '-u', 'root', 'ghcr.io/idaholab/malcolm/opensearch:24.12.0', '/usr/share/opensearch/bin/opensearch-keystore', 'create']() returned 0: ['opensearch', 'uid=1000(opensearch) gid=1000(opensearch) groups=1000(opensearch)', 'Created opensearch keystore in /usr/share/opensearch/config/opensearch.keystore', '']
Ensuring "/home/tlacuache/devel/data/Malcolm/zeek/current" exists
Ensuring "/home/analyst/Malcolm/nginx/ca-trust" exists
Ensuring "/home/analyst/Malcolm/netbox/media" exists
...
```

So, a few lines before the "Ensuring ... exists" stuff, you'll see something about creating the opensearch.keystore file. Can you share that output?

**Comment by mmguero** [comment]
I'm going to convert this from an issue to a discussion in the Troubleshooting section.

**Comment by jvlavl** [comment]
Yes, i'm using the ./scripts/start command

The opensearch is the latest version from the from the docker hub and not the one 
[ghcr.io/idaholab/malcolm/opensearch:24.12.0](http://ghcr.io/idaholab/malcolm/opensearch:24.12.0)

I will send the output ./scripts/start -v and compare it the output in this email.
[output.txt](https://github.com/user-attachments/files/18034610/output.txt)

thank you

**Comment by mmguero** [comment, marked as answer]
Ahhh, wait a second... interesting. you're just running the [hedgehog profile](https://cisagov.github.io/Malcolm/docs/live-analysis.html#Profiles) (for forwarding to another Malcolm instance), NOT the full Malcolm profile? that might be the issue, it's probably not creating that file for the hedgehog profile, even though it's not really used the docker-compose wants it to be there.

I think I can fix it, but in the meantime if that's actually what you're intending to be running you could probably just `touch opensearch/opensearch.keystore` to work around it.

**Reply by jvlavl** [reply]
Hello,

Yes, you are correct I just want to run the hedghog profile.
This workarround is working touch opensearch/opensearch.keystore.

I'm still testing further.  

In the mean time I found out that the port on my switch that i'm using for mirroring is interrupting the flows what is causing that radio and IPTV is dropping. 

I want to put the hedghog between my internet router and my internal network and make the hedghog the gateway.

ISP <--> hedghog <--> Switch <--> clients
                                                          AP
                                                           Malcolm 

Rgds,

John

**Reply by mmguero** [reply]
Opened #533 to track the issue with the opensearch.keystore not being created. I'm not able to offer you help in making your hedgehog a gateway, but it's basically a standard linux debian distribution so you should have flexibility to do what you need to do.
